source/basic.tex

```diff
@@ -575,22 +575,22 @@
 \item default template argument\iref{temp.param}
 \end{itemize}
 in a program provided that
-each definition appears in a different translation unit, and
+each definition appears in a different translation unit and
 the definitions satisfy the following requirements.
 Given such an entity \tcode{D} defined in more than one translation unit,
 for all definitions of \tcode{D}, or,
 if \tcode{D} is an unnamed enumeration,
-for all definitions of \tcode{D} that are reachable at each program point,
+for all definitions of \tcode{D} that are reachable at any given program point,
 the following requirements shall be satisfied.
 \begin{itemize}
-\item Each such definition of \tcode{D}
-shall not be attached to a named module\iref{module.unit}.
-\item Each such definition of \tcode{D} shall consist of
+\item Each such definition
+shall be attached to the global module\iref{module.unit}.
```

Comment:
Default [template] arguments aren't attached to any module, because they aren't entities, despite the introductory lie "Given such an entity `D`". To the extent that this is a normative change (which, given said lie, is debatable), I think it disallows repeating default [template] arguments in any context, which seems undesirable.

Reply:
Maybe we can rectify (or at least reduce) the lie and clarify this bullet at the same time?  Something like "Given such an entity or default (template) argument _D_" and then change the bullet in question here to "If _D_ is an entity, each such definition&hellip;".

Reply:
This takes us into problematic grounds: an editorial change that moves from "wording is ambiguous and intent is unclear" to "wording is unambiguous and we have no idea if it matches the intent". In particular: should default (template) arguments be repeatable in multiple TUs if the entity to whom they are providing a default is attached to a named module?

I think we should move this to a core issue. (Obviously the removing the unnecessary repetitive "of `D`" and other cleanups are fine to do editorially.)

Reply:
I think the "current intent" (_i.e._, old status quo plus what has been _discussed_ and voted on) is clear that such default arguments can be repeated, but I'm glad to leave it to a Core issue to decide whether that intent is bad.  The next revision here will do that.